### PR TITLE
Fixed CI step order and made setup scripts more robust

### PR DESCRIPTION
**Pull request by adal-chiriliuc-reef**

### CODE REVIEW

File: {{cookiecutter.repostory_name}}/.github/workflows/ci.yml [hunk 1]
```diff
@@ -39,14 +39,16 @@ jobs:
...
       - name: Install dependencies
         run: python -m pip install --upgrade nox 'pdm>=2.12,<3'
+      - name: Prepare environment
+        run: ./setup-dev.sh
       - name: Run dockerized services
         run: docker compose up -d --wait
       - name: Run migrations
         run: cd app/src && pdm run python manage.py wait_for_database --timeout 120 && pdm run python manage.py migrate
+      - name: Run static file collection
+        run: cd app/src && pdm run python manage.py collectstatic
```
Comment (mjurbanski-reef) [inline {{cookiecutter.repostory_name}}/.github/workflows/ci.yml:51]: why is this needed?

File: {{cookiecutter.repostory_name}}/setup-dev.sh [hunk 2]
```diff
@@ -1,25 +1,39 @@
...
 ENV_DIR="./envs/dev"
 # shellcheck disable=SC2164
 cd "${PROJECT_DIR}"
 
-# Create a lock file, install Python dependencies
-[ -f pdm.lock ] || pdm lock --group :all
+# Workaround for PDM which sometimes creates a 3.10 venv
+if [[ ! -d ".venv" ]]; then
+    python3.11 -m venv .venv
+fi
```
Comment (mjurbanski-reef) [inline {{cookiecutter.repostory_name}}/setup-dev.sh:14]: please put pdm issue link here so we can know when we can remove this workaround

File: {{cookiecutter.repostory_name}}/.github/workflows/ci.yml (hunk 1 above)
Comment (adal-chiriliuc-reef) [reply {{cookiecutter.repostory_name}}/.github/workflows/ci.yml:51]: Some tests fail without this step with this error:
`ValueError: Missing staticfiles manifest entry for 'img/favicon.png'`

https://github.com/reef-technologies/reef.pl/actions/runs/8616637087/job/23614679098

Comment (adal-chiriliuc-reef) [reply {{cookiecutter.repostory_name}}/.github/workflows/ci.yml:51]:
```python
=========================== short test summary info ============================
FAILED reefpl/github/tests.py::GitHubTests::test_account_name_null - ValueError: Missing staticfiles manifest entry for 'img/favicon.png'
FAILED reefpl/github/tests.py::GitHubTests::test_authentication_error - ValueError: Missing staticfiles manifest entry for 'img/favicon.png'
FAILED reefpl/github/tests.py::GitHubTests::test_login - ValueError: Missing staticfiles manifest entry for 'img/favicon.png'
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 3 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
```

But I have removed these tests anyway, since they are just copy-pasted from `django-allauth`, and fail even in the old `reef.pl` (before `cruft update`). 

So now we don't need `collectstatic` anymore, so we can remove it then? I'm not sure.

File: {{cookiecutter.repostory_name}}/setup-dev.sh (hunk 2 above)
Comment (adal-chiriliuc-reef) [reply {{cookiecutter.repostory_name}}/setup-dev.sh:14]: I searched and I could find any relevant issue. Created one now: https://github.com/pdm-project/pdm/issues/2789

File: {{cookiecutter.repostory_name}}/.github/workflows/ci.yml (hunk 1 above)
Comment (mjurbanski-reef) [reply {{cookiecutter.repostory_name}}/.github/workflows/ci.yml:51]: as for manifest thing, I think it occurs there since 'django.contrib.staticfiles.storage.ManifestStaticFilesStorage' is used there, so unless we want to introduce it into cookiecutter as well then I don't see reason for a fix for it to be introduced in cookiecutter and not there alone.

We can discuss private repo changes in that repo PR alone.
Here lets make sure we are addressing only problems that are common to all projects using cookiecutter.

File: {{cookiecutter.repostory_name}}/setup-dev.sh (hunk 2 above)
Comment (mjurbanski-reef) [reply {{cookiecutter.repostory_name}}/setup-dev.sh:14]: Please put the link in workaround comment

Comment (adal-chiriliuc-reef) [reply {{cookiecutter.repostory_name}}/setup-dev.sh]: done

File: {{cookiecutter.repostory_name}}/.github/workflows/ci.yml (hunk 1 above)
Comment (adal-chiriliuc-reef) [reply {{cookiecutter.repostory_name}}/.github/workflows/ci.yml]: Sorry about that, I forgot this is a public repo.

I have reverted the `collectstatic` step, indeed it is not required for all projects.

**Review by mjurbanski-reef — APPROVED** [review]
